HolidayHat/Program.cs

```diff
@@ -1,71 +1,103 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Diagnostics;
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
-
-namespace HolidayHat
+﻿using System.Diagnostics;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+var oldResults = ReadResultsFile();
+var newResults = AssignNewRecipients(oldResults);
+WriteResultsFile(newResults);
+ShowResultsMessage();
+
+
+static Results AssignNewRecipients(Results oldResults)
+{
+    Results? results = null;
+    while (results == null)
+        results = TryAssignNewRecipients(oldResults);
+    return results;
+}
+
+static Results? TryAssignNewRecipients(Results oldResults)
 {
-    class Program
+    var gifters = new List<Person>(people);
+    var recipients = new List<Person>(people);
+    var random = new Random();
+    var results = new Results();
+
+    foreach (var gifter in gifters)
     {
-        enum Person
+        var invalidRecipients =
+            new List<Person>() { gifter, oldResults[gifter] };
+        var validRecipients = recipients.Except(invalidRecipients);
+
+        if (!validRecipients.Any())
+            return null;
+
+        Person recipient;
+        do
         {
-            Christie,
-            Ron,
-            Dianne,
-            John,
-            Tisa
+            var randomIndex = random.Next(recipients.Count);
+            recipient = recipients[randomIndex];
```

Comment:
Would it work to use `validRecipients` instead of `recipients` on line 38-39? Seems like then we might be able to remove the do-while.